Userland/Libraries/LibWeb/DOM/ARIARoleNames.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022, Jonah Shafran <jonahshafran@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/DOM/ARIARoleNames.h>
+
+namespace Web::DOM::ARIARoleNames {
+
+#define __ENUMERATE_ARIA_ROLE(name) FlyString name;
+ENUMERATE_ARIA_ROLES
+#undef __ENUMERATE_ARIA_ROLE
+
+[[gnu::constructor]] static void initialize()
+{
+    static bool s_initialized = false;
+    if (s_initialized)
+        return;
+
+#define __ENUMERATE_ARIA_ROLE(name) name = #name;
+    ENUMERATE_ARIA_ROLES
+#undef __ENUMERATE_ARIA_ROLE
+
+    // Special case for C++ keyword
+    switch_ = "switch";
+    s_initialized = true;
+}
+
+bool is_abstract_role(FlyString const& role)
+{
+    return role.is_one_of(
+        ARIARoleNames::command,
+        ARIARoleNames::composite,
+        ARIARoleNames::input,
+        ARIARoleNames::landmark,
+        ARIARoleNames::range,
+        ARIARoleNames::roletype,
+        ARIARoleNames::section,
+        ARIARoleNames::sectionhead,
+        ARIARoleNames::select,
+        ARIARoleNames::structure,
+        ARIARoleNames::widget,
+        ARIARoleNames::window);
+}
+
+bool is_non_abstract_aria_role(FlyString const& role)
```

Comment:
The spec separates these out into smaller categories. Maybe the code could match that too, with `is_widget_role()`, `is_document_structure_role()` etc. Then this function could just return `is_widget_role(role) || is_document_structure_role(role) || ...`